### DISCUSSION TITLE
Scorer vs metrics/test cases

### DISCUSSION THREAD

**Discussion by mausch** [confident-ai/deepeval#1219] [Q&A] [answered]
Hi, I'm new to deepeval, currently evaluating it.
One thing I find puzzling is the API divide between e.g. https://docs.confident-ai.com/docs/metrics-hallucination and `Scorer.hallucination_score`
I get that one uses an LLM and the other one doesn't, but it's pretty confusing to have these two wildly different APIs? Is there a reason for this?

**Comment by penguine-ip** [comment, marked as answer]
Hey @mausch please consider anything that is not documented deprecated or for internal use :)

The scorers were there because in the past it was relied for scoring metrics but since we've moved to 100% LLM as a judge they are still there in case users want to build their own custom metrics here: https://www.deepeval.com/guides/guides-building-custom-metrics#building-a-custom-non-llm-eval

**Reply by mausch** [reply]
Thanks, so is `Scorer.hallucination_score` deprecated? or for internal use? (which one?)
I don't follow, if writing a custom eval means potentially using this how can it be deprecated or for internal use?

**Reply by penguine-ip** [reply]
Deprecated for metrics, but not for using it on its own. Deprecated for metrics because we don't recommend it, but you're free to use it for custom evals if you insist.
